lib/facter/systemd.rb

```diff
@@ -50,14 +50,10 @@
 end
 
 Facter.add(:systemd_internal_services) do
+  require 'json'
   confine systemd: true
   setcode do
-    command_output = Facter::Util::Resolution.exec(
-      'systemctl list-unit-files --no-legend --no-pager "systemd-*" -t service --state=enabled,disabled,enabled-runtime,indirect'
-    )
-    lines = command_output.lines.lazy.map { |line| line.split(%r{\s+}) }
-    lines.each_with_object({}) do |(service, status, *), result|
-      result[service] = status
-    end
+    command_output = Facter::Util::Resolution.exec('systemctl list-unit-files "systemd-*" -t service --state=enabled,disabled,enabled-runtime,indirect -o json')
```

Comment:
I didn't use it in the past because not all systemctl versions support the option. Do you know when the option was introduced?

Reply:
google told me: version 239 (May 2017). maybe i drop this PR. if we still have to support older systemds the more "hacky" version was a better approach for this 🤔

Reply:
may i wait until the older osses are dropped 😆 
https://github.com/voxpupuli/puppet-systemd/pull/496

Reply:
EL8 will not be dropped with that patch and EOL for EL8 is 2029.

Reply:
ah damnit okay... then i think i drop it for the moment. the current implementation is working for older and newer versions of systemd. didnt know the feature is this "new" 😆